### PR TITLE
[Feature: #163966307] New Password Component

### PR DESCRIPTION
**Pull request by ebenezerdon**
#### What does this PR do?
- This PR completes the feature for `Password Reset`

#### Description of Task to be completed?
- When a user clicks on the link sent to his/her email, they should be redirected to a `new password` page where they can reset their password.

#### How should this be manually tested?
- deploy this branch `Feature/163966307/new-password-component`
- navigate to `/resetpassword`
- enter a valid email address
- you should recieve a mail with a link
- click the link, and you should be redirected to a `New Password` page
- fill the form with matching `passwords`
- your password reset should be successful

#### What are the relevant pivotal tracker stories?
- [#163966307](https://www.pivotaltracker.com/story/show/163966307)

### CODE REVIEW

File: src/app/App.jsx [hunk 1]
```diff
@@ -22,6 +23,7 @@ const App = () => (
         <Route path="/subject" component={Subject} exact />
         <Route path="/articles" component={Article} exact />
         <Route path="/resetpassword" component={ResetPassword} exact />
+        <Route path="/newpassword" component={NewPassword} exact />
         <Route path="/create-article" component={CreateArticle} exact />
```
Comment (rajeman) [inline src/app/App.jsx:27]: @ebenezerdon I noticed I can navigate to this route even without requesting for password change. I think that should not be possible.

Comment (ebenezerdon) [reply src/app/App.jsx:27]: I think so too. Do you know how I can go about preventing that?

Comment (rajeman) [reply src/app/App.jsx:27]: You can make the route check if there is a valid resetToken passed to it as a query. If not, it should redirect to some other page. There is a route on our backend for validating resetToken.
There is likely a better way though.

Comment (ebenezerdon) [reply src/app/App.jsx:27]: Okay, that should work. We'll create a story for it then.

**Comment by rajeman** [comment]
Why create a new story?

**Comment by ebenezerdon** [comment]
It doesn't seem like a quick `one-line` solution. I feel it will be better off on a different story since it involves making another `API` call. The **PRs** should be easier to review that way.
Also, I'm currently working on displaying articles on our frontend.

**Comment by andela-mabdussalam** [comment]
I do not think we need to validate the token before we render the component. If the page was requested wrongly, it would mean no token is sent and the backend manages the validation.

Merging as it is.

**Review by andela-mabdussalam — APPROVED** [review]

**Pull request merged**